lib/opensusebasetest.pm

```diff
@@ -548,7 +548,8 @@ sub wait_boot {
     if (check_var('BACKEND', 'ipmi')) {
         select_console 'sol', await_console => 0;
         # boot from harddrive
-        assert_screen([qw(virttest-pxe-menu qa-net-selection prague-pxe-menu pxe-menu)], 200);
+        assert_screen([qw(virttest-pxe-menu qa-net-selection prague-pxe-menu pxe-menu linux-login)], 200);
```

Comment:
The only thing I don't like about this solution is that we actually handle all of these in `sub reconnect_mgmt_console`.
What's about adding all missing tags there and removing this clause completely, so we assert `linux-login` for textmode installations on L641? WDYT?

Reply:
You mean removing the `if` block completely when `BACKEND=ipmi`? This way it'll call `wait_grub` which I think could work with `scooter-1.qa.suse.de` but not with `holmes.qa.suse.de` (I don't see a grub screen in holmes' video for which to assert). I will push the change to test and add the results here.

Keep in mind though that both verification runs in the description are tests with Gnome. `linux-login` is shown in SOL before the job changes focus to `x11`, so I don't think we should treat them as `textmode` which we would be doing with the assert in L641

Reply:
Removed the `if` block for IPMI. Testing with:

https://openqa.suse.de/tests/3241678 (sles4sap_scc_gnome_sap_inst_wizard)

https://openqa.suse.de/tests/3241679 (default@64bit-ipmi)

Reply:
Good point, then it definitely belongs to `reconnect_mgmt_console`, for the remote backends we handle grub there too, so `first_boot` asserts only expected login console or x11, depending on the setup. We currently have same issue on spvm, where use ssh console, and we don't have adjustments yet to connect over vnc after reboot.

Reply:
Both runs failed.

* sles4sap_scc_gnome_sap_inst_wizard@holmes: is failing while asserting `linux-login` or `emergency-shell`, so this looks better than I expected. Seems to be doing the assert in L647, but Issue here seems to be timeout for this particular system.

* default@scooter-1.qa.suse.de: fails while asserting `grub2`, but `grub2` was already asserted in `grub_test`

Reply:
Decided to:

1. Exclude `boot/grub_test` when `BACKEND=ipmi`. As it is, `installation/first_boot` will call `wait_boot` on such scenario, which in turns calls to `wait_grub` in most conditions. Excluding `boot/grub_test` moves the `assert_screen` for `grub2` to `installation/first_boot`. Will wait for https://openqa.suse.de/tests/3242731 to finish to trigger a verification run for that scenario with this PR.

2. Remove the IPMI `if` block from `wait_boot` so it asserts the login screen in L641 as suggested, and also added a `select_console 'sol'` before that call to `assert_screen` to ensure the test is connected to a valid SOL session. Verification run for the sles4sap scenario with holmes.qa.suse.de it's in https://openqa.suse.de/tests/3247249

WDYT @rwx788 ?

Reply:
Verification run for default@64bit-ipmi: https://openqa.suse.de/tests/3250831